src/sql/workbench/contrib/extensions/browser/scenarioRecommendations.ts

```diff
@@ -128,7 +128,7 @@ export class ScenarioRecommendations extends ExtensionRecommendations {
 		if (!scenarioType) {
 			return Promise.reject(new Error(localize('scenarioTypeUndefined', 'The scenario type for extension recommendations must be provided.')));
 		}
-		return this.promptedExtensionRecommendations.filterIgnoredOrNotAllowed(this.productService.recommendedExtensionsByScenario[scenarioType] || [])
+		return this.promptedExtensionRecommendations?.filterIgnoredOrNotAllowed(this.productService.recommendedExtensionsByScenario[scenarioType] || [])
```

Comment:
Yeah, but why was this removed and what ramifications does this have? We should understand what filterIgnoredOrNotAllowed is doing and whether it's something we actually need.

There's been some major reworking of this stuff by vs code so let's make sure that we're still adhering to the rules of the service and doing the correct thing

https://github.com/microsoft/vscode/commit/b6117be33574480076fcadd8ef169a2b73a5ab1a#diff-b4dba09ee7f635c62719cf2808c8f973ddbb8cebc66960ea15cc5c3036e369c1

Reply:
We can probably get rid of the `promptedExtensionRecommendations` stuff completely looking at the vs code stuff. They removed it and currently none of our stuff uses it.

But we should still make sure that the ignore extension recommendation stuff works as expected as well (i.e. are our custom recommendation types doing the right thing here to filter?)

Reply:
Yeah, or Extension Management component is highly diverged from vscode source and has major merge conflicts every large integration.  We need to rework our code to be more in-line with current vscode implementation, while maintaining our forked behaviors.  The next merge already has major conflicts here again, so I may try to do some of the refactoring in the next merge.